packages/block-library/src/search/edit.js

```diff
@@ -424,13 +424,11 @@ export default function SearchEdit( {
 							}
 							step={ 1 }
 							onChange={ ( newWidth ) => {
-								const filteredWidth =
-									widthUnit === '%' &&
-									parseInt( newWidth, 10 ) > 100
-										? 100
-										: newWidth;
+								const parsedNewWidth = parseInt( newWidth, 10 );
 								setAttributes( {
-									width: parseInt( filteredWidth, 10 ),
+									width: Number.isNaN( parsedNewWidth )
+										? undefined
+										: parsedNewWidth,
```

Comment:
A bit nitty, but this makes it read like `parsedNewWidth` can be an _invalid_ value (`NaN`), when in fact it was a valid value (empty string, aka `undefined`).

I think making that explicit in the parsing stage will avoid raising alarm bells for the reader. For example:

```jsx
const parsedNewWidth =
	newWidth === ''
		? undefined
		: parseInt( newWidth, 10 );
setAttributes( { width: parsedNewWidth } );
```

Reply:
Makes sense, I'll apply the suggestion and merge